src/services/SessionConfigRequestProcessor.ts

```diff
@@ -0,0 +1,60 @@
+import { Response } from "../utils/Response";
+import { CicService } from "./CicService";
+import { Metrics, MetricUnits } from "@aws-lambda-powertools/metrics";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { AppError } from "../utils/AppError";
+import { HttpCodesEnum } from "../utils/HttpCodesEnum";
+import { createDynamoDbClient } from "../utils/DynamoDBFactory";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { Constants } from "../utils/Constants";
+
+const SESSION_TABLE = process.env.SESSION_TABLE;
+
+export class SessionConfigRequestProcessor {
+	private static instance: SessionConfigRequestProcessor;
+
+	private readonly logger: Logger;
+
+	private readonly metrics: Metrics;
+
+	private readonly cicService: CicService;
+
+	constructor(logger: Logger, metrics: Metrics) {
+		if (!SESSION_TABLE) {
+			logger.error("Environment variable SESSION_TABLE is not configured", {
+				messageCode: MessageCodes.MISSING_CONFIGURATION,
+			});
+			throw new AppError( "Service incorrectly configured", HttpCodesEnum.SERVER_ERROR);
+		}
+		this.logger = logger;
+		this.metrics = metrics;
+		this.cicService = CicService.getInstance(SESSION_TABLE, this.logger, createDynamoDbClient());
+	}
+
+	static getInstance(logger: Logger, metrics: Metrics): SessionConfigRequestProcessor {
+		if (!SessionConfigRequestProcessor.instance) {
+			SessionConfigRequestProcessor.instance = new SessionConfigRequestProcessor(logger, metrics);
+		}
+		return SessionConfigRequestProcessor.instance;
+	}
+
+	async processRequest(sessionId: string): Promise<Response> {
+
+		const session = await this.cicService.getSessionById(sessionId);
+
+		if (session != null) {
```

Comment:
We have a couple of specific null checks in this file, is there a reason that we've gone for this over implicitly checking for truthy values that would catch `undefined` as well?

Reply:
Which other checks are you referring to? in here we are checking if we get a session object for the given session-id else throw an error.

Reply:
Referring to this `session != null`, and on line 51 `session?.journey != null`. 

I see why are you checking for `null` but I'm wondering why we aren't checking for all falsy values by using `if (session)` and `session?.journey ?? Constants.FACE_TO_FACE_JOURNEY`

Reply:
Done, Thanks.